src/juicy-popover.html

```diff
@@ -163,23 +162,59 @@
             var expandable = this.shadowRoot.querySelector("#expandable");
+            var expandableSlot = null;
+            [].forEach.call( expandable.children, function(el) {
+                if(el.assignedNodes && el.assignedNodes()[0]){
+                    expandableSlot = el.assignedNodes()[0];
+                }
+                if(el.getDistributedNodes && el.getDistributedNodes()[0]){
+                    expandableSlot = el.getDistributedNodes()[0];
+                }
+            });
+            var styleTarget = expandable;
+            var nonStyleTarget = expandableSlot;
+            if (!expandable.clientWidth) {
+                styleTarget = expandableSlot;
+                nonStyleTarget = expandable;
+            }
             var elem = this.expandedOnElement || this.shadowRoot.querySelector("#handle");
+            while (!(elem.getAttribute('slot') && elem.getAttribute('slot') == "handle")) {
+                if(elem.parentElement) {
+                    elem = elem.parentElement;
+                } else {
+                    elem = this.expandedOnElement || this.shadowRoot.querySelector("#handle");
+                    break;
+                }
+            }
             var position = this.getAttribute("position") || "bottom left";
 
             if (position.indexOf("top") > -1) {
-                expandable.style.top = elem.offsetTop + "px";
+                styleTarget.style.top = elem.offsetTop - styleTarget.offsetHeight + "px";
             } else {
-                expandable.style.top = elem.offsetTop + elem.offsetHeight + "px";
+                styleTarget.style.top = elem.offsetTop + elem.offsetHeight - nonStyleTarget.offsetTop + "px";
             }
-
+    
             if (position.indexOf("right") > -1) {
-                expandable.style.left = elem.offsetLeft + elem.offsetWidth + "px";
-            } else {
-                expandable.style.left = elem.offsetLeft + "px";
+                styleTarget.style.right = "";
+                styleTarget.style.marginRight = "";
+            } else if(position.indexOf("left") > -1){
+                styleTarget.style.right = "0";
+                styleTarget.style.marginRight = "-" + elem.clientWidth + "px";
+            } else if(position.indexOf("auto") > -1){
+                if(window.innerWidth - elem.getBoundingClientRect().right < styleTarget.offsetWidth + (parseInt(styleTarget.style.marginRight, 10) || 0)) {
+                    styleTarget.style.right = "0";
+                    styleTarget.style.marginRight = "-" + elem.clientWidth + "px";
+                } else {
+                    styleTarget.style.right = "";
+                    styleTarget.style.marginRight = "";
+                }
             }
         };
 
         JuicyPopoverElementPrototype.expand = function (elem) {
+            if (!elem.assignedSlot && this.attachShadow) {
+                elem = elem.parentElement;
+            }
```

Comment:
Why this is needed?

Reply:
You're right, I forgot to erase to it.
FIxed in 913fd7233b3029a9113e958c0d50458e814d2ac9 .